bitsandbytes/backends/cpu_xpu_common.py

```diff
@@ -181,7 +182,7 @@ def igemmlt_impl(A, B, SA=None, SB=None, out=None, Sout=None, dtype=torch.int32)
     A_reshaped = A.reshape(m, k)
 
     # torch._int_mm is available on CPU since torch 2.4
-    if _torch_version_prereq(2, 4):
+    if _torch_version_prereq(2, 4) and A.device.type == "cpu":
         C = torch._int_mm(A_reshaped, B.T).to(dtype)
     else:
         C = torch.matmul(A_reshaped.float(), B.t().float()).to(dtype)
```

Comment:
In IPEX, there is support for int8 input and int8 weight matmul https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-gpu/blob/master/csrc/gpu/aten/operators/QLinear.cpp#L96. Why, in the case where an int8 QLinear implementation already exists, did we choose to use the method of input.dequant * weight.dequant for computation? Is it because the input/weight tensor representations in b&b and torch are incompatible?

Reply:
Hi @zhuyuhua-v . I didn't notice it. Please reach me via Teams so we can discuss it.

Reply:
We have aligned that the IPEX API is not compatible with BNB usage, IPEX will give an int8 gemm op just like cpu in the future.

Reply:
cool, makes sense, thanks for clarifying!